backendCore/common/ai_drive.py

```diff
@@ -0,0 +1,272 @@
+import json
+import traceback
+import urllib.request as request
+import csv
+import numpy as np
+import pandas as pd
+
+from sklearn.datasets import *
+from sklearn.model_selection import train_test_split
+import torch
+from torch.autograd import Variable
+from torch.utils.data import TensorDataset, DataLoader
+
+from backendCore.common.constants import CSV_FILE_PATH, DEFAULT_DATASETS, ONNX_MODEL
+from backendCore.dl.dl_model import DLModel
+from backendCore.dl.dl_model_parser import parse_deep_user_architecture
+from backendCore.dl.dl_trainer import train_deep_model
+
+
+def read_dataset(url):
```

Comment:
We should probably refactor this behavior. If we are given a url to a csv dataset, we should be able to:

1. parse the csv file name from the url itself (or use a generic name that the user knows about)
2. get the csv contents and build temporary file (but do not store it locally)
3. send the temporary csv file to s3

Do you think that you can do this refactor or we should have another dev do it? I think this refactor can be done as an atomic task? @dwu359